controllers/lvmcluster_controller_test.go

```diff
@@ -43,23 +50,79 @@ var _ = Describe("LVMCluster controller", func() {
 					DeviceClasses: []lvmv1alpha1.DeviceClass{{Name: "test"}},
 				},
 			}
-			Expect(k8sClient.Create(ctx, lvmCluster)).Should(Succeed())
 
-			//Check that the status.Ready field has been set to true. This is a placeholder test and will
-			// be modified to check for the actual resources once they are implemented.
+			// CSI Driver Resource
+			csiDriverName := types.NamespacedName{Name: TopolvmCSIDriverName}
+			csiDriverOut := &storagev1.CSIDriver{}
+
+			// Topolvm Controller Deployment
+			controllerName := types.NamespacedName{Name: TopolvmControllerDeploymentName, Namespace: testLvmClusterNamespace}
+			controllerOut := &appsv1.Deployment{}
+
+			By("Indicate setting CR status to be ready after CR is deployed")
+			Expect(k8sClient.Create(ctx, lvmClusterIn)).Should(Succeed())
 
-			lvmClusterLookupName := types.NamespacedName{Name: testLvmClusterName, Namespace: testLvmClusterNamespace}
-			lvmCluster1 := &lvmv1alpha1.LVMCluster{}
+			// placeholder to check CR status.Ready field to be true
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, lvmClusterName, lvmClusterOut)
+				if err != nil {
+					return false
+				}
+				return lvmClusterOut.Status.Ready
+			}, timeout, interval).Should(Equal(true))
+
+			// present of csi driver resource
+			By("Confirming csi driver resource is present")
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, csiDriverName, csiDriverOut)
+				if err != nil {
+					return false
+				}
+				return true
+			}, timeout, interval).Should(BeTrue())
 
+			// presence of topolvm controller deployment
+			By("Confirming topolvm deployment is present")
 			Eventually(func() bool {
-				err := k8sClient.Get(ctx, lvmClusterLookupName, lvmCluster1)
+				err := k8sClient.Get(ctx, controllerName, controllerOut)
 				if err != nil {
 					return false
 				}
-				return lvmCluster1.Status.Ready
+				return true
 			}, timeout, interval).Should(BeTrue())
-			// Let's make sure our Schedule string value was properly converted/handled.
-			Expect(lvmCluster1.Status.Ready).Should(Equal(true))
+
+			By("Confirming deletion of lvm cluster")
```

Comment:
Please move the deletion to a separate test.

Reply:
- this is all part of `It("Reconciles an LvmCluster, "` if I were to move it to a separate test then I think I need to deploy again and then delete